userland/examples/ble_adv/main.c

```diff
@@ -0,0 +1,27 @@
+#include <stdio.h>
+#include "ble.h"
+#include <timer.h>
+
+int main(void)
+{
+  printf("\r\ndemo app\n");
+  unsigned char name[] = "TockOS";
+  unsigned char data[] = "1337";
+  unsigned char tx[] = {0x1};
+
+  printf("return for ble_adv_set_power %d\r\n", ble_adv_set_txpower(0));
+  printf("return for ble_adv_set_interval %d\r\n", ble_adv_set_interval(0));
+  
+  // name and data are strings, remove \0 by subtracting 1
+  ble_adv_data(BLE_HS_ADV_TYPE_COMP_NAME, sizeof(name) - 1, name);
+  ble_adv_data(BLE_HS_ADV_TYPE_MFG_DATA, sizeof(data) - 1, data);
+  ble_adv_data(BLE_HS_ADV_TYPE_TX_PWR_LVL, 1, tx);
```

Comment:
Yes, the capsule will validate that only 30 bytes of payload but at the moment it is not possible to reset the payload. (1 more byte it possible though however we not tested it extensively yet)

@alevy @brghena do you want a separate system call to reset the payload?

Further, ble_adv_data is performed in a separate allow perhaps it could be improved by combining several AD Types in one allow call.